### DISCUSSION TITLE
How to create separate partitons for kernel slots on system.conf for RAUC?

### DISCUSSION THREAD

**Discussion by TulasiGanapareddy** [rauc/rauc#1369] [Q&A]
Hii all..
    I don't have any idea about how to create the separate kernel slots on RAUC. I used the yocto project and the system is raspberrypi-3. It was created the rootfs-A & B slots. my partitions are look like this on system.conf

``` ini
[system]
compatible=RaspberryPi3
bootloader=uboot

[keyring]
path=/etc/rauc/ca.cert.pem

[slot.rootfs.0]
device=/dev/mmcblk0p2
type=ext4
bootname=A

[slot.rootfs.1]
device=/dev/mmcblk0p3
type=ext4
bootname=B
```

Please help me how to do?
Please, I will kindly appreciate your help.

Regards,
Tulasi

**Comment by ejoerns** [comment]
In the config, this could look like

``` ini
[slot.kernel.0]
device=/dev/mmcblk0p4
type=raw
parent=rootfs.0

[slot.kernel.1]
device=/dev/mmcblk0p5
type=raw
parent=rootfs.1
```

The actual partitioning needs to be done in your `wic` file or whatever technique you use for creating.

For creating an appropriate bundle using the meta-rauc `bundle.bbclass`, you would need to add a slot image like

``` bitbake
RAUC_BUNDLE_SLOTS += "kernel"
RAUC_SLOT_kernel = "linux-raspberrypi"
RAUC_SLOT_kernel[type] = "kernel"
```

**Reply by TulasiGanapareddy** [reply]
Thank-you for your response.

**Reply by TulasiGanapareddy** [reply]
[slot.kernel.0]
device=/dev/mmcblk0p4
type=raw
parent=rootfs.0

[slot.kernel.1]
device=/dev/mmcblk0p5
type=raw
parent=rootfs.1

You mentioned that above parent=rootfs.0 for the kernel how to do it without assign the parent

**Reply by ejoerns** [reply]
The parent relation is required to update the redundant kernel together with the rootfs in the same slot group.

Why do you want to omit the parent relation? Is there no fixed relation between the kernel and the rootfs in your setup?

**Reply by TulasiGanapareddy** [reply]
Okay, I understood i will create with parent relation 
Thank you for your response.
